docs/programmers_guide/witness_formal_spec.md

```diff
@@ -17,13 +17,14 @@ In the end, when there are no more instructions left, there MUST be only one ite
 
 ### Nodes
 
-There are multiple types of nodes that can be present on the stack.
-
-- `hashNode`
-- `branchNode`
-- `accountNode`
-- `valueNode`
-- `extensionNode`
+```
+type Node = nil
+          | HashNode (raw_hash:Byte[32])
```

Comment:
Perhaps, instead of introducing types like Byte[32], and Byte[], simply use Infinite integer type and just explain that we treat it as (mod 2^256) in some cases, and just a sequence of bytes in other cases (conveniently, value is always trimmed from the leading zeros)

Reply:
sounds good, will make the notion more compact

Reply:
on the second though, won't it be more readable for developers to keep these types? 
sure, we can just call them `Hash` and `ByteArray` to make it more readable.

We can then describe them in words -- 
"`Hash` is 32-byte value, representing Keccak256 hash of a value"
"`ByteArray` is a sequence of bytes of an arbitrary size"